source/containers/document-pii-detection/parsers/image_analysis/general_ocr/ocr_main.py

```diff
@@ -0,0 +1,386 @@
+        img = img.copy()
+        ort_inputs = {self.ort_session.get_inputs()[0].name: img}
+        preds = {}
+        preds['maps'] = self.ort_session.run(None, ort_inputs)[0]
+
+        post_result = self.postprocess_op(preds, shape_list)
+        dt_boxes = post_result[0]['points']
+        if self.det_algorithm == "SAST" and self.det_sast_polygon:
+            dt_boxes = self.filter_tag_det_res_only_clip(dt_boxes, ori_im.shape)
+        else:
+            dt_boxes = self.filter_tag_det_res(dt_boxes, ori_im.shape)
+        return dt_boxes
+
+class TextRecognizer():
+    def __init__(self, model_path):
+        self.weights_path = model_path + 'rec_standard.onnx'
+
+        self.limited_max_width = 1280
+        self.limited_min_width = 16
+
+        self.rec_image_shape = [3, 32, 320]
+        self.character_type = 'ch'
+        self.rec_batch_num = 6
+        self.rec_algorithm = 'CRNN'
+        self.use_zero_copy_run = False
+        postprocess_params = {
+            'name': 'CTCLabelDecode',
+            "character_type": 'ch',
+            "character_dict_path": model_path + 'keys_v1.txt',
+            "use_space_char": True
+        }
+        self.postprocess_op = build_post_process(postprocess_params)
+
+        self.ort_session = onnxruntime.InferenceSession(self.weights_path, providers=['CUDAExecutionProvider'] if cuda_available else ['CPUExecutionProvider'])
+        _ = self.ort_session.run(None, {"backbone": np.zeros([1, 3, 32, 64], dtype='float32')})
+
+    def resize_norm_img(self, img, max_wh_ratio):
+        imgC, imgH, imgW = self.rec_image_shape
+        assert imgC == img.shape[2]
+        if self.character_type == "ch":
+            imgW = int((32 * max_wh_ratio))
+        imgW = max(min(imgW, self.limited_max_width), self.limited_min_width)
+        h, w = img.shape[:2]
+        ratio = w / float(h)
+        ratio_imgH = math.ceil(imgH * ratio)
+        ratio_imgH = max(ratio_imgH, self.limited_min_width)
+        if ratio_imgH > imgW:
+            resized_w = imgW
+        else:
+            resized_w = int(math.ceil(imgH * ratio))
+        resized_image = np.array(Image.fromarray(img).resize((resized_w, imgH)))
+        #resized_image = cv2.resize(img, (resized_w, imgH))
+        resized_image = resized_image.astype('float32')
+        resized_image = resized_image.transpose((2, 0, 1)) / 255
+        resized_image -= 0.5
+        resized_image /= 0.5
+        padding_im = np.zeros((imgC, imgH, imgW), dtype=np.float32)
+        padding_im[:, :, 0:resized_w] = resized_image
+        return padding_im
+
+    def __call__(self, img_list):
+        img_num = len(img_list)
+        # Calculate the aspect ratio of all text bars
+        width_list = []
+        for img in img_list:
+            width_list.append(img.shape[1] / float(img.shape[0]))
+        # Sorting can speed up the recognition process
+        indices = np.argsort(np.array(width_list))
+
+        # rec_res = []
+        rec_res = [['', 0.0]] * img_num
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are initializing a multidimensional array using replication. Replicating a `list` with `*` creates references to the existing objects, not copies, which could introduce bugs. We recommend that you create a `list` of the desired length and then fill in each element with a newly created `list`, or use `list` comprehension. For more information about `list` comprehension, see the following.

[Learn more](https://docs.python.org/3/faq/programming.html#how-do-i-create-a-multidimensional-list)